account_ecotax/tests/test_ecotax.py

```diff
@@ -0,0 +1,544 @@
+# Copyright 2016-2023 Akretion France
+#   @author: Alexis de Lattre <alexis.delattre@akretion.com>
+# Copyright 2021 Camptocamp
+#   @author Silvio Gregorini <silvio.gregorini@camptocamp.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from random import choice
+
+from odoo.tests.common import Form, tagged
+
+from odoo.addons.account.tests.common import AccountTestInvoicingCommon
+
+
+@tagged("-at_install", "post_install")
+class TestInvoiceEcotaxe(AccountTestInvoicingCommon):
+    @classmethod
+    def setUpClass(cls, chart_template_ref=None):
+        super().setUpClass(chart_template_ref)
+        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
+
+        # ACCOUNTING STUFF
+        # 1- Tax account
+        cls.invoice_tax_account = cls.env["account.account"].create(
+            {
+                "code": "47590",
+                "name": "Invoice Tax Account",
+                "account_type": "liability_current",
+                "company_id": cls.env.user.company_id.id,
+            }
+        )
+        cls.invoice_ecotax_account = cls.env["account.account"].create(
+            {
+                "code": "707120",
+                "name": "Ecotax Account",
+                "account_type": "income",
+                "company_id": cls.env.user.company_id.id,
+            }
+        )
+        # 2- Invoice tax with included price to avoid unwanted amounts in tests
```

Comment:
We need a set of tests with taxes not included in price. The default tax in France in Odoo is a 20% G which is not included in price, switching this is not an option for FR users.